architecture/azure/azure-abac-auth.md

```diff
@@ -0,0 +1,374 @@
+
+![Azure Entra ID Attributes](attribute-values-examples.png "width=800")
+
+- **NServiceBus Message**: The message body and headers are a crucial PIP, providing context about the action and its parameters. For example:
+
+snippet: abac-pip-as-nservicebus-behavior
+
+- **Azure Blob Storage Index Tags**: A PIP that provides attributes about the resource itself.
+
+![Azure Storage BLOB Index](azure-abac-storage-blob-index.png "width=800")
+
+**PAP (Policy Administration Point)** - The PAP is the interface or system where policies and attributes are managed.  
+
+- **The Azure Portal**: The primary PAP for managing Azure's native ABAC. This is where you assign roles with conditions and define custom security attributes.
+
+![Azure Portal Add Attributes](project-attribute-add.png "width=800")
+
+## Application
+
+Below are example scenarios where Azure ABAC could be used to solve the challenges with RBAC in distributed systems using NServiceBus, native ABAC support for Azure Storage Queues and Azure Blob Storage, and Azure Entra ID with security attributes.
+
+- [Enforcing Data Residency with Region-Specific Queues](/architecture/azure/azure-abac-auth.md#application-scenario-enforcing-data-residency-with-region-specific-queues)
+- [Just-in-Time Check for Purchase Approval Authority](/architecture/azure/azure-abac-auth.md#application-scenario-just-in-time-check-for-purchase-approval-authority)
+- [Attribute-Scoped Log Processing](/architecture/azure/azure-abac-auth.md#application-scenario-attribute-scoped-log-processing-with-nservicebus)
+
+### Scenario: Enforcing Data Residency with Region-Specific Queues
+
+This scenario ensures that data subject to residency laws is only processed by systems located in the correct geographical region.
+
+**Solves**:
+
+- ⚠️ An unmanageable proliferation of fine-grained access roles
+
+**By using:**
+
+- ✅ Azure Entra ID
+- ✅ Azure Storage Queues
+- ✅ Azure ABAC with Custom Security Attributes
+- ✅ NServiceBus Endpoints and Messages
+
+**The Problem ⚠️**: A global company uses a single system to process customer orders. An order from France must be processed by a service running in a European data center. A new queue is created for each region processor. The company will end up managing **N** individual permissions for **N** processors. This could potentially scale up in other scenarios to hundreds, leading to an _unmanageable proliferation of fine-grained access roles_ to accommodate complex distributed systems.
+
+**The Solution** ✅: The solution is to route messages to region-specific Azure Storage Queues and use native Azure ABAC to strictly enforce that only regional processor endpoints can access their corresponding queue. This is achieved by giving all processors permissions to all queues and conditionally allowing these processes access to queues based on their region attribute. This allows for a single permission to be applied for **N** processors and queues. For example:
```

Comment:
A couple of thoughts here:
- We try to use full sentences as opposed to how it's currently framed
- I don't think we use emojis anywhere in the documentation
- Generally, I think it's best to lead with the problem if we want to keep this framing.